pages/profile/[username].tsx

```diff
@@ -78,6 +79,16 @@ const UserProfile: React.FC = () => {
     }
   })
 
+  const lessonStatus: LessonStatus[] = _.get(data, 'userInfo.lessonStatus', [])
+  const profileStars: Star[] = []
+  lessonStatus.forEach(({ starsReceived }) => {
```

Comment:
You are using the `profileStars` variable and the `forEach` to do a combination of reducing and filtering. You should separate out this logic, and use `filter` along with a more appropriate higher order function so that your code becomes more explicit and clear about what is happening. It can be made even more simple by removing the nested `forEach` inside with a spread operator instead.

```
const validProfiles = lessonStatus.filter(({ starsReceived }) => starsReceived)
const profileStars: Star[] = validProfiles.reduce((acc, { starsReceived }) => {
  acc.push(...starsReceived)
  return acc
}, [])
```

Reply:
Ah, I see you changed your code before I finished making this comment. Lemme look the changes now